### ISSUE TITLE
Issue on GitHub Actions with latest Zulu 8u352 JDK

### ISSUE THREAD

**Issue by arcuri82** [alibaba/java-dns-cache-manipulator#115]
Hi,
  thanks for developing and maintaining this library. 

Today, all of a sudden, my tests on GitHub Actions started to fail, due to `java-dns-cache-manipulator`, we get the following:
```
java.lang.NoSuchFieldException: addressCache
       java.lang.Class.getDeclaredField(Class.java:2070)     
       com.alibaba.dcm.internal.InetAddressCacheUtilForJava8.getAddressCacheAndNegativeCacheOfInetAddress0(InetAddressCacheUtilForJava8.java:179)
       com.alibaba.dcm.internal.InetAddressCacheUtilForJava8.getAddressCacheOfInetAddress(InetAddressCacheUtilForJava8.java:155)
       com.alibaba.dcm.internal.InetAddressCacheUtilForJava8.clearInetAddressCache(InetAddressCacheUtilForJava8.java:294)
       com.alibaba.dcm.DnsCacheManipulator.clearDnsCache(DnsCacheManipulator.java:257)
```

after some investigation, the issue seems related to update of JDK in GitHub Action. By default, it uses Azul Zulu builds. In its most recent 8u352 version (which can be downloaded [here](https://www.azul.com/downloads/?package=jdk)) they changed the code of `InetAddress`.

Where before (eg, OpenJDK 8u275) it was:
```
private static Cache addressCache = new Cache(Cache.Type.Positive);
```

now it is:
```
private static final ConcurrentMap<String, Addresses> cache =   new ConcurrentHashMap<>();
```

I guess `DnsCacheManipulator` could be modified to check for `cache`  if it gets a `NoSuchFieldException` on  `addressCache`

**Comment by arcuri82** [comment]
thx! btw, it is not about holding off upgrading, as that is the current default in GitHub Action when you use:

```
      - name: Setup JDK 1.8
        uses: actions/setup-java@v1
        with:
          java-version: 1.8
```

**Comment by oldratlee** [comment]
> thx! btw, it is not about holding off upgrading, as that is the current default in GitHub Action when you use:
> 
> ```yml
>       - name: Setup JDK 1.8
>         uses: actions/setup-java@v1
>         with:
>           java-version: 1.8
> ```

How about setting a more precise `java-version`(e.g. `8.0.345`)?  @arcuri82 

I have tried this, it seems working:

- commit: https://github.com/alibaba/java-dns-cache-manipulator/commit/cb5ca861fd168086e481f3ec28bb39c4fb96823a#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR13
- GitHub Action job `JDK 8.0.345`
https://github.com/alibaba/java-dns-cache-manipulator/actions/runs/3297915524/jobs/5439344792#step:4:10

--------------

This leads to create a commit that shouldn't be needed. 😔

**Comment by lameiro** [comment]
[This recent commit](https://github.com/openjdk/jdk8u-dev/commit/222305888bda3f0e225839d832cdf8983996a229) in OpenJDK8 changed InetAddress, it no longer has the addressCache/negativeCache fields.

**Comment by GOODBOY008** [comment]
@oldratlee I had compare different version of jdk1.8.
![image](https://user-images.githubusercontent.com/13617900/206332617-aba979cd-5944-4386-b951-c9f0d2c89722.png)
![image](https://user-images.githubusercontent.com/13617900/206332632-c2ef6403-3cd8-4dcf-b63b-cfdc04a294f8.png)

**Comment by oldratlee** [comment]
@arcuri82 @lameiro @GOODBOY008 

fix this issue in release `v1.8.0`:
https://github.com/alibaba/java-dns-cache-manipulator/releases/tag/v1.8.0

``` xml
<dependency>
    <groupId>com.alibaba</groupId>
    <artifactId>dns-cache-manipulator</artifactId>
    <version>1.8.0</version>
</dependency>
```

please upgrade version and check. 💕

**Comment by arcuri82** [comment]
@oldratlee thx! seems working ;)
